ui/app/AppLayouts/Wallet/panels/SwapInputPanel.qml

```diff
@@ -58,6 +58,7 @@ Control {
     readonly property int rawValueMultiplierIndex: amountToSendInput.multiplierIndex
     readonly property bool valueValid: amountToSendInput.inputNumberValid
     readonly property bool amountEnteredGreaterThanBalance: value > maxSendButton.maxSafeValue
+    readonly property string accountBalanceFormatted: root.currencyStore.formatCurrencyAmount(d.maxCryptoBalance, d.selectedHolding.symbol)
```

Comment:
this being exposed from here is rather weird, this is an account property should not really come from the SwapInputPanel, but from the adaptor instead.

Reply:
Yeah, was just discussing that with @alexjba ; looks like we have an issue there with the `ModelEntry` (pointing to the currently selected account) which returns old/incorrect values

Reply:
TLDR, I agree :)

I'll do some experiment/test case that exhibits that faulty behavior so that Alex can have a look at it; meanwhile I'd keep this (with a TODO/FIXME) as the `ModelEntry` approach was just returning wrong results for me (I had this code originally there)

Reply:
ok the please please create a GH issue and link it here :)

Reply:
@alexjba see the video here:

[Záznam obrazovky z 2024-07-12 00-56-19.webm](https://github.com/user-attachments/assets/528f3cf8-2ed8-4311-91c8-da85221b8889)

The code is in SwapModalAdaptor, I even added the helper for chainId:
```qml
...
FastExpressionRole {
  name: "accountBalance"
  expression: {
    root.swapFormData.selectedNetworkChainId
    return d.processAccountBalance(model.address)
  }
  expectedRoles: ["address"]
  }
...
ModelEntry {
    id: selectedAccountEntry
    sourceModel: root.nonWatchAccounts
    key: "address"
    value: root.swapFormData.selectedAccountAddress
}
```

and then in SwapModal:
```qml
SwapApproveCapModal {
...
accountBalanceFormatted: root.swapAdaptor.selectedAccount.accountBalance.formattedBalance
```

Simply shows the "old" value from the previously selected network

Reply:
Not sure it's the `ModelEntry` or because we calculate it using a function which uses the old/deprecated `SubmodelProxyModel`

Reply:
I created a separate issue for this under: https://github.com/status-im/status-desktop/issues/15554

Reply:
thanks @caybro can you please link the task in the code?